### ISSUE TITLE
git completion broken when using git-dir and work-tree

### ISSUE THREAD

**Issue by Amarandus** [zimfw/zimfw#171]
Hi,

tl;dr: git completion for git add is problematic when using bare repos with a work tree in different directories. 

I am managing my dotfiles as described in https://developer.atlassian.com/blog/2016/02/best-way-to-store-dotfiles-git-bare-repo/ 
For this, I need to use `git` with `--git-dir` and `--work-tree`.
When trying to perform tab completion, zim does prevent me from using it, because (logically) I am not in a git repository. 
Nontheless, I want it to give me a file completion (considering that there is no sensible way to give any other useful tab completion).

How do I handle this? Is there a way to exclude my alias from the git completion?

```
Zim commit ref:   dca1803
Zsh version:      zsh 5.3.1 (x86_64-unknown-linux-gnu)
System info:      Linux skybox 4.10.11-1-ARCH #1 SMP PREEMPT Tue Apr 18 08:39:42 CEST 2017 x86_64 GNU/Linux
```
Gist for .zimrc/.zshrc:
https://gist.github.com/Amarandus/f801b2786189ea29075502e4df176a8c

**Comment by ericbn** [comment]
Can you please provide exact steps to reproduce (in terms of a sequence of shell commands) and expected results?

**Comment by ericbn** [comment]
This seems to be an upstream issue. As @Eriner recommended in #283:

>  If you'd like to see changes in this regard, take a look here: `/usr/share/zsh/functions/Completion/Unix/_git` and hit up the zsh mailing list.
